packages/publish-views/lib/shacl.js

```diff
@@ -0,0 +1,44 @@
+import through2 from 'through2'
+import { sh } from '@tpluscode/rdf-ns-builders'
+import { ValidationError } from './ValidationError.js'
+
+const REPORTS_KEY = 'COMBINED_SHACL_REPORTS'
+
+export function combineShaclReports({ context, report }) {
+  const variables = context.variables.has(REPORTS_KEY)
+    ? context.variables
+    : context.variables.set(REPORTS_KEY, [])
+
+  variables.get(REPORTS_KEY).push(report)
+
+  // continue stream
+  return true
+}
+
+export function failOnAnyViolations() {
+  const { variables } = this
+  const ignoreWarnings = variables.get('ignoreWarnings')
+
+  return through2.obj(function (chunk, _, next) {
+    this.push(chunk)
+    next()
+  }, function (done) {
+    const reports = variables.get(REPORTS_KEY)
+    if (reports?.some(hasViolations(ignoreWarnings))) {
+      this.destroy(new ValidationError(reports))
+    }
+
+    done()
+  })
```

Comment:
This step is basically a no-op. The flush callback destroys the stream if there are error in the end and also has the option to ignore the reports if they contain only `sh:Warning`